### PR TITLE
loosen oauth2 dependency, release version 1.2

### PR DESCRIPTION
**Pull request by linberg**
This pull request aims to release a new (lower) version of this gem which will be compatible with the `auth` upgrade.

### Manual Testing

* Use a local version of this gem
* update `config/omniauth_providers.yml` in `auth` (ask me for credentials)
* set `fake_auth=false` in your KP app of choice (I used `beratung`)
* try to login and logout using KP to authenticate

### CODE REVIEW

File: omniauth-kaeuferportal.gemspec [hunk 1]
```diff
@@ -2,8 +2,8 @@
 require File.expand_path('../lib/omniauth-kaeuferportal/version', __FILE__)
 
 Gem::Specification.new do |gem|
-  gem.add_dependency 'omniauth', '~> 1.1.0'
-  gem.add_dependency 'oauth2', '=0.7.1'
+  gem.add_dependency 'omniauth', '~> 1.0'
+  gem.add_dependency 'oauth2', '~> 1.1.0'
```
Comment (mbd-s) [inline omniauth-kaeuferportal.gemspec]: I'm curious if we need to be so specific with the version here. Assuming `oauth2` follows SemVer, `~> 1.1` _should_ be enough to keep us safe.

**Comment by mbd-s** [comment]
What do you think about creating a CHANGELOG?

**Comment by mbd-s** [comment]
We should probably be adding tags (again) so it's easier to work with specific versions of the gem: https://help.github.com/en/articles/working-with-tags

File: omniauth-kaeuferportal.gemspec (hunk 1 above)
Comment (linberg) [reply omniauth-kaeuferportal.gemspec]: I did try a higher version at first, but unfortunately we get these errors when I leave `oauth2` at `~> 1.1` (which upgrades `oauth2` to `1.4.1` in addition to a major upgrade of [jwt](https://github.com/jwt/ruby-jwt) from `1.5.1` to `2.2.1`)

```Der Parameter 'redirect_url' is ungültig. Er muss zu der URL des anfragenden Services passen. Bitte gib eine gültige URL an, zu der umgeleitet werden soll.```

and

```(kaeuferportal) Authentication failure! invalid_credentials: OAuth2::Error, Der Parameter 'client_id' fehlt. Bitte gib die ID der Anwendung an.```

**Pull request merged**